akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -307,6 +306,46 @@ object Source {
   def queue[T](bufferSize: Int, overflowStrategy: OverflowStrategy): Source[T, SourceQueue[T]] =
     new Source(scaladsl.Source.queue[T](bufferSize, overflowStrategy).mapMaterializedValue(new SourceQueueAdapter(_)))
 
+  /**
+    * Creates Source that works as dispatcher for legacy blocking IO API
+    *
+    * Example:
+    * {{{
+    * Source.blocking(
+    *   () -> new BufferedReader(new FileReader("...")),
+    *   reader -> reader.readLine(),
+    *   reader -> reader.close())
+    * }}}
+    *
+    * You can use supervision strategy to handle exceptions for `read` function. All exceptions thrown by `create`
+    * or `close` will complete the stream.
+    *
+    * `Restart` supervision strategy will close and create blocking IO again. Default strategy is `Stop` which means
+    * that stream will be terminated on error in `read` function by default.
+    *
+    *  Stream calls close and completes when `read` returns None.
+    */
+  def unfoldResource[T, S](create: function.Creator[S],
+                     read: function.Function[S, Option[T]],
+                     close: function.Procedure[S]): javadsl.Source[T, NotUsed] =
+    new Source(scaladsl.Source.unfoldResource[T,S](create.create, read.apply, close.apply))
+
+  /**
+    * Creates Source that works as dispatcher for legacy blocking IO API.
+    * It's similar to `unfoldResource` but returns `Futures` instead of plain values.
+    *
+    * You can use supervision strategy to handle exceptions for `read` function or failures of produced `Futures`.
+    * All exceptions thrown by `create` or `close` as well as fails of returned futures will complete the stream.
+    *
+    * `Restart` supervision strategy will close and create resource. Default strategy is `Stop` which means
+    * that stream will be terminated on error in `read` function (or future) by default.
+    *
+    *  Stream calls close and completes when `read` future success with None.
+    */
+  def unfoldResourceAsync[T, S](create: function.Creator[Future[S]],
+    read: function.Function[S, Future[Option[T]]],
+    close: function.Function[S, Future[Done]]): javadsl.Source[T, NotUsed] =
```

Comment:
Yeah, for some reason I missed this activity of transforming all Futures to CompletionStage for java API